lib/authenticated-api/authenticated-api.ts

```diff
@@ -19,11 +21,11 @@ const DEFAULT_API_LATENCY_THRESHOLD = cdk.Duration.minutes(1);
 const DEFAULT_LAMBDA_DURATION_THRESHOLD = cdk.Duration.minutes(1);
 
 export class AuthenticatedApi extends cdk.Construct {
+  readonly httpApi: apigatewayv2.HttpApi;
   readonly apiId: string;
   readonly httpApiId: string;
   readonly domainName: apigatewayv2.DomainName;
 
-  private httpApi: apigatewayv2.HttpApi;
```

Comment:
I exposed the ApiGateway here so we could something like we do here in the application code consuming this construct. However, it's not required for the purpose of this so I don't know if it's needed?

Reply:
When we discussed this in the past - the original idea was to keep things in the construct. Shaun thought this would allow us to enforce things, and exposing them out of the construct would allow people to set things we didn't want them too. E.g. minimum memory size on a lambda.

We changed the Lambda worker to create a Lambda Worker Function which could be accessed outside, for example to allow it to be used for permission, but still wrapped the CDK class.

Exposing this here does seem to break the original idea of encapsulation, but the way the CDK works, maybe that encapsulation isn't going to work.

So I'm in two minds as to whether I think this is a good idea. 

Falling back to "it's easier to make things public than to make things private once they have been public" - if it's not required - is it worth not making it public for now?

Reply:
I made the change early when started working on this but I agree there's no need to expose it - I undid it.